src/org/exist/http/servlets/HttpServletRequestWrapper.java

```diff
@@ -332,14 +273,15 @@ public String getParameter(final String name) {
 
     @Override
     public BufferedReader getReader() throws IOException {
-        if (contentBodyRecorded()) {
-            String encoding = request.getCharacterEncoding();
-            if (encoding == null) {
-                encoding = "UTF-8";
-            }
-            return new BufferedReader(new InputStreamReader(getContentBodyInputStream(), encoding));
-        } else {
-            return request.getReader();
+        String encoding = request.getCharacterEncoding();
+        if (encoding == null) {
+            encoding = "UTF-8";
```

Comment:
use literal from NIO ?

Reply:
Not much point here, as HttpServletRequest works with a String